scripts/tutorial-uploader/lib/local-tutorial-data.ts

```diff
@@ -0,0 +1,73 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import yaml from "js-yaml";
+import { readFile } from "fs/promises";
+import { dirname, join } from "path";
+import { isString, isNumber, isBoolean } from "lodash";
+
+/* Information specified in the YAML file */
+export interface LocalTutorialData {
+  title: string;
+  short_description: string;
+  slug: string;
+  status: string;
+  local_path: string;
+  category: string;
+  topics: string[];
+  reading_time: number;
+  catalog_featured: boolean;
+}
+
+function relativiseLocalPath(
+  tutorial: LocalTutorialData,
+  path: string,
+): LocalTutorialData {
+  tutorial.local_path = join(dirname(path), tutorial.local_path);
+  return tutorial;
+}
+
+export async function readTutorialData(
+  path: string,
+): Promise<LocalTutorialData[]> {
+  const raw = await readFile(path, "utf8");
+  const parsed = yaml.load(raw) as unknown[];
+  return parsed
+    .map((i) => verifyLocalTutorialData(i))
```

Comment:
I like how you are verifying if the object matches the interface with `verifyLocalTutorialData` giving a detailed error for the attributes that fail. 

Optional suggestion: I think we could make it more explicit that the function is also type-casting the objects into `LocalTutorialData` to make the code more readable and less prone to errors if in the future we need to change the verification somehow.

One option could be to change the function's name. Alternatively, we could also change the `verifyLocalTutorialData` function to return `void` and move the check before this return using a `forEach`. With this strategy the second map could to be `.map((i) => relativiseLocalPath(i as LocalTutorialData, path));`

Reply:
Good idea, thanks! Done in ae35611a332d77b925b2d0d65ff43615a168afd5.